### PR TITLE
KAFKA-5182: Reduce session timeout in request quota test

### PR DESCRIPTION
**Pull request by rajinisivaram**
Reduce rebalance and session timeouts for join requests to trigger throttling in the request quota test.

### CODE REVIEW

**Comment by rajinisivaram** [comment]
@ijuma Can you review, please? It is a small change. Thank you!

File: core/src/test/scala/unit/kafka/server/RequestQuotaTest.scala [hunk 1]
```diff
@@ -217,9 +218,9 @@ class RequestQuotaTest extends BaseRequestTest {
           new requests.FindCoordinatorRequest.Builder(FindCoordinatorRequest.CoordinatorType.GROUP, "test-group")
 
         case ApiKeys.JOIN_GROUP =>
-          new JoinGroupRequest.Builder("test-join-group", 10000, "", "consumer",
+          new JoinGroupRequest.Builder("test-join-group", 200, "", "consumer",
             List( new JoinGroupRequest.ProtocolMetadata("consumer-range",ByteBuffer.wrap("test".getBytes()))).asJava)
```
Comment (ijuma) [inline core/src/test/scala/unit/kafka/server/RequestQuotaTest.scala]: Nit: comma missing after `,` and extra space after `List(`.

Comment (ijuma) [reply core/src/test/scala/unit/kafka/server/RequestQuotaTest.scala]: Also, as discussed offline, there are some unused `builder` fields in this class. I had removed them in a separate PR, but it makes more sense to do it here, probably. As part of that, we can simplify the code a little. Here's what I had done (ignore the `Resource => _, ResourceType => _` change):

https://github.com/ijuma/kafka/commit/49f4b0cf37f8ca396e6e90f015299d12d211932b

**Review by ijuma — COMMENTED** [review]
@rajinisivaram, to make sure I understand, is rebalancing taking too long (potentially due to KIP-134) and hence no throttling happens? So, by reducing the timeouts, it means that rebalancing times out?

One other thing to consider is reducing `group.initial.rebalance.delay.ms`.

File: core/src/test/scala/unit/kafka/server/RequestQuotaTest.scala (hunk 1 above)
Comment (rajinisivaram) [reply core/src/test/scala/unit/kafka/server/RequestQuotaTest.scala]: @ijuma Thank you, fixed these.

**Comment by rajinisivaram** [comment]
@ijuma I had missed the `group.initial.rebalance.delay.ms` change, which is probably the reason why the failures are occurring in the build. I was thinking of subsequent join requests rather than the first one, which need low timeouts to trigger throttling. I have changed the code to set `group.initial.rebalance.delay.ms=0` as well. Thanks.

File: core/src/test/scala/unit/kafka/server/RequestQuotaTest.scala [hunk 2]
```diff
@@ -197,29 +185,29 @@ class RequestQuotaTest extends BaseRequestTest {
...
-          new requests.FindCoordinatorRequest.Builder(FindCoordinatorRequest.CoordinatorType.GROUP, "test-group")
+          new FindCoordinatorRequest.Builder(FindCoordinatorRequest.CoordinatorType.GROUP, "test-group")
 
         case ApiKeys.JOIN_GROUP =>
-          new JoinGroupRequest.Builder("test-join-group", 10000, "", "consumer",
-            List( new JoinGroupRequest.ProtocolMetadata("consumer-range",ByteBuffer.wrap("test".getBytes()))).asJava)
-           .setRebalanceTimeout(60000)
+          new JoinGroupRequest.Builder("test-join-group", 200, "", "consumer",
+            List(new JoinGroupRequest.ProtocolMetadata("consumer-range", ByteBuffer.wrap("test".getBytes()))).asJava)
+           .setRebalanceTimeout(100)
```
Comment (dguy) [inline core/src/test/scala/unit/kafka/server/RequestQuotaTest.scala:210]: Do we need to change this if `GroupInitialRebalanceDelayMsProp` is set to 0? The group should rebalance immediately unless another group member joins before `InitialDelayedJoin.onComplete` is called. 
I guess there is no harm in setting it to 100

Comment (rajinisivaram) [reply core/src/test/scala/unit/kafka/server/RequestQuotaTest.scala:210]: @dguy @ijuma We send this request (for the same group) many times in a loop until throttling occurs. The delay for the first join request is based on `GroupInitialRebalanceDelayMsProp`, but delays for subsequent join requests to the same group are based on `rebalanceTimeout`.

**Review by ijuma — APPROVED** [review]
LGTM. I think it would be slightly cleaner to use a different client id per request, but this is simpler and seems to work. If we have additional failures, we can revisit.

**Comment by rajinisivaram** [comment]
@ijuma @dguy Thank you for the reviews. Merging to trunk.

**Pull request closed without merging**